server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3865,10 +3881,10 @@ private EngineConfig newEngineConfig(LongSupplier globalCheckpointSupplier) thro
             circuitBreakerService,
             globalCheckpointSupplier,
             replicationTracker::getRetentionLeases,
-            () -> getOperationPrimaryTerm(),
+            this::getOperationPrimaryTerm,
             tombstoneDocSupplier(),
             isReadOnlyReplica,
-            replicationTracker::isPrimaryMode,
+            this::isStartedPrimary,
```

Comment:
Lets rename the argument name from `primaryModeSupplier` to `isStartedPrimary` as well in `EngineConfigFactory` .